### PR TITLE
PMM-6568 new modal

### PR DESCRIPTION
**Pull request by lunaticusgreen**
[PMM-6568](https://jira.percona.com/browse/PMM-6568)

### CODE REVIEW

File: pmm-app/src/shared/components/Elements/Modal/Modal.styles.ts [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+      width: 100%;
+      max-height: calc(90vh - ${theme.spacing.d} * 2);
+  `,
+  modalHeaderClose: css`
+      height: 100%;
+      display: flex;
+      align-items: center;
+      flex-grow: 1;
+      justify-content: flex-end;
+      padding-right: ${theme.spacing.sm};
```
Comment (atymchuk) [inline pmm-app/src/shared/components/Elements/Modal/Modal.styles.ts:52]: I guess it should be a margin - padding impacts the click area, which would be disproportionate in this case.

File: pmm-app/tests/QAN/pages/qanPage.js [hunk 2]
```diff
@@ -64,7 +64,7 @@ module.exports = {
     showSelectedDisabled: '//button[@data-qa="qan-filters-show-selected" and @disabled ]',
     table: '//table//tr[2]',
     tableRow: 'div.tr',
-    noQueries: "//h1[contains(text(), 'No queries available for this combination of filters')]",
+    noQueries: '//h1[contains(text(), \'No queries available for this combination of filters\')]',
```
Comment (atymchuk) [inline pmm-app/tests/QAN/pages/qanPage.js:67]: Let's make sure we avoid the escaped quotes `\'` - replace them with double quotes `"`. It should work.
```suggestion
    noQueries: '//h1[contains(text(), "No queries available for this combination of filters")]',
```

File: pmm-app/tests/pages/dashboardPage.js [hunk 3]
```diff
@@ -543,8 +543,8 @@ module.exports = {
     clearSelection: '//a[@ng-click="vm.clearSelections()"]',
     Last2Days: '//span[contains(text(), \'Last 2 days\')]',
     timeRangePickerButton: '.btn.navbar-button.navbar-button--tight',
-    rootUser: "//div[contains(text(), 'root')]",
-    dataLinkForRoot: "//div[contains(text(), 'Data links')]/..//a",
+    rootUser: '//div[contains(text(), \'root\')]',
```
Comment (atymchuk) [inline pmm-app/tests/pages/dashboardPage.js:546]: Ditto for the quotes.

File: pmm-app/src/shared/components/Elements/Modal/Modal.tsx [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+
+export const Modal: FC<ModalWindow> = (props) => {
+  const {
+    isVisible, children, title, onClose, closeOnClickaway = true, closeOnEscape = true
+  } = props;
+  const theme = useTheme();
+  const styles = getStyles(theme);
+
+  if (closeOnEscape) {
+    useEffect(() => {
```
Comment (atymchuk) [inline pmm-app/src/shared/components/Elements/Modal/Modal.tsx:23]: How about this rule from https://reactjs.org/docs/hooks-rules.html#only-call-hooks-at-the-top-level?
`Don’t call Hooks inside loops, conditions, or nested functions`. 

I wonder, doesn't eslint complain about that?

**Review by atymchuk — CHANGES_REQUESTED** [review]

File: pmm-app/src/shared/components/Elements/Modal/Modal.tsx (hunk 4 above)
Comment (lunaticusgreen) [reply pmm-app/src/shared/components/Elements/Modal/Modal.tsx:23]: Good point, fixed

**Review by atymchuk — APPROVED** [review]

**Pull request merged**